include/boost/gil/image_processing/numeric.hpp

```diff
@@ -203,6 +127,150 @@ inline void generate_gaussian_kernel(boost::gil::gray32f_view_t dst, double sigm
     }
 }
 
+/// \brief Generates Sobel operator in horizontal direction
+/// \ingroup ImageProcessingMath
+///
+/// Generates a kernel which will represent Sobel operator in
+/// horizontal direction of specified degree (no need to convolve multiple times
+/// to obtain the desired degree).
+/// https://www.researchgate.net/publication/239398674_An_Isotropic_3_3_Image_Gradient_Operator
+template <typename T = float, typename Allocator = std::allocator<T>>
+inline kernel_2d<T, Allocator> generate_dx_sobel(unsigned int degree = 1)
```

Comment:
I started thinking if `numeric.hpp` bag should be split into multiple specialised headers then this would go to e.g. `sobel.hpp`

However, such refactoring may be depending on the merging of numeric into core. 
@stefanseefeld what's the status of this plan? ->https://lists.boost.org/boost-gil/2019/09/0296.php

Reply:
I believe I will be doing that after full cleanup, like moving everything to kernel_2d, adding more tests, etc. If you don't mind that approach, could you please close the issue?

Reply:
...or `sobel.hpp`, `scharr.hpp`, ...

Yes, sounds good. Resolving this part.